### ISSUE TITLE
Compilation error on a fresh clone

### ISSUE THREAD

**Issue by fbdo** [poutsma/web-function-sample#4]
Hi!

Sorry for the maybe lame question, but I saw this example and tried to execute in my machine and on a fresh clone, when I do a simple mvn package I see the following error:

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.5.1:compile (default-compile) on project web-function-sample: Compilation failure
[ERROR] C:\dev\workspace\web-function-sample\src\main\java\org\springframework\samples\web\reactive\function\Server.java:[98,23] error: incompatible types: ReactorHttpHandlerAdapter cannot be converted to Function<? super HttpChannel,? extends Publisher<Void>>

Basically the error is in this line of Sever.java:

server.startAndAwait(adapter);

As the project uses a fixed version of the reactor-netty library, I suppose something changed in the spring-web lib, as the project is set up to use the latest version.

Meanwhile I will try to figure out what changed, I just wanted to let you know.

Regards.

**Comment by yatsykve** [comment]
@fbdo please let us know if you find how to fix this error.
@poutsma do you have any updates on this example?

**Comment by yatsykve** [comment]
ok, I found some workaround, we can switch `spring-web-reactive` to version `M3` :
```
<dependency>
    <groupId>org.springframework</groupId>
    <artifactId>spring-web-reactive</artifactId>
    <version>5.0.0.M3</version>
</dependency>
```

**Comment by poutsma** [comment]
This should be fixed now.
